src/main/java/clippy/command/Parser.java

```diff
@@ -0,0 +1,120 @@
+        case "unmark":
+            // todo: check if second argument is a valid number
+            return dispatch(CommandType.UNMARK, new String[] { args[1] });
+        case "todo":
+            if (args.length < 2) {
+                throw new ClippyTodoEmptyDescriptionException();
+            }
+            return dispatch(CommandType.TODO, new String[] { args[1] });
+        case "deadline":
+        {
+            int byIndex = command.indexOf("/by ");
+            if (byIndex == -1 || command.length() < byIndex + 4) {
+                throw new ClippyMissingDeadlineException();
+            }
+            String description = command.substring(9, byIndex).trim();
+            String byDateString = command.substring(byIndex + 4).trim();
+            try {
+                LocalDate byDate = LocalDate.parse(byDateString);
+                return dispatch(CommandType.DEADLINE, new String[] { description }, byDate);
+            } catch (DateTimeParseException e) {
+                throw new ClippyInvalidDateException();
+            }
+        }
+        case "event": {
+
+            int fromIndex = command.indexOf("/from ");
+            int toIndex = command.indexOf("/to ");
+            if (fromIndex == -1 || toIndex == -1 || toIndex - fromIndex < 5 ||
+                    command.length() - toIndex < 4) {
+                throw new ClippyInvalidEventException();
+            }
+
+            String description = command.substring(6, fromIndex).trim();
+            String fromDateString = command.substring(fromIndex + 6, toIndex).trim();
+            String toDateString = command.substring(toIndex + 4).trim();
+            try {
+                LocalDate fromDate = LocalDate.parse(fromDateString);
+                LocalDate toDate = LocalDate.parse(toDateString);
+                return dispatch(CommandType.EVENT, new String[] { description }, fromDate, toDate);
+            } catch (DateTimeParseException e) {
+                throw new ClippyInvalidDateException();
+            }
+
+        }
+        case "list":
+            return dispatch(CommandType.LIST, EMPTY_ARG_LIST);
+        case "find":
+            // todo: check if args[1] is valid input
+            return dispatch(CommandType.FIND, new String[] { args[1].trim() });
+        case "delete":
+            // todo: check for valid list index
+            return dispatch(CommandType.DELETE, new String[]{ args[1] });
+        default:
+            throw new ClippyUnknownCommandException();
+        }
+    }
+
+    /**
+     * Instantiates the correct command handler object to be returned, assuming
+     * inputs are all valid.
+     *
+     * @param commandType The type of command given.
+     * @param args The arguments to the command.
+     * @param dates Dates, if any, to be used in the command.
+     * @return
+     */
+    private static Command dispatch(CommandType commandType, String[] args, LocalDate... dates) {
```

Comment:
I like your idea of creating a dispatch() method to cooperate with your parse method, but will it become a better encapsulation if Parser class is not under the packet command?